### PR TITLE
Add Makefile

### PR DESCRIPTION
**Pull request by cemkeylan**
I think a Makefile is overall better for installing across systems. It also eliminates the requirement of the package maintainer to check for file changes, as it is controlled from the source repository Makefile. I am open to your comments!

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# kiss
+# See LICENSE file for copyright and license details
```
Comment (dylanaraps) [inline Makefile]: These two lines can be removed.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+PREFIX=/usr/local
+MANPREFIX=${PREFIX}/share/man
+
+all: options
+
+options:
+	@echo "PREFIX = ${PREFIX}"
+	@echo "MANPREFIX = ${MANPREFIX}"
```
Comment (dylanaraps) [inline Makefile]: This can be removed.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,60 @@
+# kiss
+# See LICENSE file for copyright and license details
+
+PREFIX=/usr/local
```
Comment (dylanaraps) [inline Makefile]: Change `PREFIX`  to `/usr`.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+all: options
+
+options:
+	@echo "PREFIX = ${PREFIX}"
+	@echo "MANPREFIX = ${MANPREFIX}"
+
+install:
+	@mkdir -p ${DESTDIR}${PREFIX}/bin
+	@mkdir -p ${DESTDIR}${MANPREFIX}/man1
+	@echo "Installing KISS to ${DESTDIR}${PREFIX}/bin"
```
Comment (dylanaraps) [inline Makefile]: This can be removed.

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	@install -Dm755 kiss ${DESTDIR}${PREFIX}/bin/kiss
+	@echo "Installing manpages to ${DESTDIR}${MANPREFIX}"
```
Comment (dylanaraps) [inline Makefile]: This can be removed.

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+	@install -Dm644 kiss.1 ${DESTDIR}${MANPREFIX}/man1/kiss.1
+
+install-utils:
+	@mkdir -p ${DESTDIR}${PREFIX}/usr/bin
+	@echo "Installing kiss-utils to ${DESTDIR}${PREFIX}/bin"
```
Comment (dylanaraps) [inline Makefile]: This can be removed.

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+install:
+	@mkdir -p ${DESTDIR}${PREFIX}/bin
+	@mkdir -p ${DESTDIR}${MANPREFIX}/man1
+	@echo "Installing KISS to ${DESTDIR}${PREFIX}/bin"
+	@install -Dm755 kiss ${DESTDIR}${PREFIX}/bin/kiss
+	@echo "Installing manpages to ${DESTDIR}${MANPREFIX}"
+	@install -Dm644 kiss.1 ${DESTDIR}${MANPREFIX}/man1/kiss.1
+
+install-utils:
+	@mkdir -p ${DESTDIR}${PREFIX}/usr/bin
```
Comment (dylanaraps) [inline Makefile]: This can be removed.

File: Makefile [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+all: options
+
+options:
+	@echo "PREFIX = ${PREFIX}"
+	@echo "MANPREFIX = ${MANPREFIX}"
+
+install:
+	@mkdir -p ${DESTDIR}${PREFIX}/bin
+	@mkdir -p ${DESTDIR}${MANPREFIX}/man1
```
Comment (dylanaraps) [inline Makefile]: These two lines can be removed.

File: Makefile [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+	@install -Dm755 contrib/kiss-manifest-tree ${DESTDIR}${PREFIX}/bin/kiss-manifest-tree
+	@install -Dm755 contrib/kiss-maintainer ${DESTDIR}${PREFIX}/bin/kiss-maintainer
+	@install -Dm755 contrib/kiss-orphans ${DESTDIR}${PREFIX}/bin/kiss-orphans
+	@install -Dm755 contrib/kiss-revdepends ${DESTDIR}${PREFIX}/bin/kiss-revdepends
+	@install -Dm755 contrib/kiss-owns ${DESTDIR}${PREFIX}/bin/kiss-owns
+	@install -Dm755 contrib/kiss-size ${DESTDIR}${PREFIX}/bin/kiss-size
+	@install -Dm755 contrib/kiss-depends ${DESTDIR}${PREFIX}/bin/kiss-depends
+
+uninstall:
+	@echo "Removing KISS from ${DESTDIR}${PREFIX}/bin"
```
Comment (dylanaraps) [inline Makefile]: This can be removed.

File: Makefile [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+	@rm -f ${DESTDIR}${PREFIX}/bin/kiss
+	@echo "Removing KISS manpage from ${DESTDIR}${MANPREFIX}"
```
Comment (dylanaraps) [inline Makefile]: This can be removed.

File: Makefile [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+	@rm -f ${DESTDIR}${MANPREFIX}/man1/kiss.1
+
+uninstall-utils:
+	@echo "Removing kiss-utils from ${DESTDIR}${PREFIX}/bin"
```
Comment (dylanaraps) [inline Makefile]: This can be removed.

File: Makefile [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+	@rm -f ${DESTDIR}${PREFIX}/bin/kiss-manifest
+	@rm -f ${DESTDIR}${PREFIX}/bin/kiss-manifest-tree
+	@rm -f ${DESTDIR}${PREFIX}/bin/kiss-orphans
+	@rm -f ${DESTDIR}${PREFIX}/bin/kiss-owns
+	@rm -f ${DESTDIR}${PREFIX}/bin/kiss-repodepends
+	@rm -f ${DESTDIR}${PREFIX}/bin/kiss-reset
+	@rm -f ${DESTDIR}${PREFIX}/bin/kiss-revdepends
+	@rm -f ${DESTDIR}${PREFIX}/bin/kiss-size
+
+.PHONY: all options install install-utils uninstall uninstall-utils
```
Comment (dylanaraps) [inline Makefile]: `options` can be removed from here.

**Comment by dylanaraps** [comment]
Thanks for the PR. :+1:.

The `mkdir` calls can be removed as the `-D` flag for `install` will create the directories automatically.

**Comment by cemkeylan** [comment]
All done!

**Comment by cemkeylan** [comment]
I also removed all, it didn't seem necessary. I can add it back if you want me to

File: Makefile [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+	@install -Dm755 contrib/kiss-repodepends ${DESTDIR}${PREFIX}/bin/kiss-repodepends
+	@install -Dm755 contrib/kiss-depends-finder ${DESTDIR}${PREFIX}/bin/kiss-depends-finder
+	@install -Dm755 contrib/kiss-export ${DESTDIR}${PREFIX}/bin/kiss-export
+	@install -Dm755 contrib/kiss-manifest-tree ${DESTDIR}${PREFIX}/bin/kiss-manifest-tree
+	@install -Dm755 contrib/kiss-maintainer ${DESTDIR}${PREFIX}/bin/kiss-maintainer
+	@install -Dm755 contrib/kiss-orphans ${DESTDIR}${PREFIX}/bin/kiss-orphans
+	@install -Dm755 contrib/kiss-revdepends ${DESTDIR}${PREFIX}/bin/kiss-revdepends
+	@install -Dm755 contrib/kiss-owns ${DESTDIR}${PREFIX}/bin/kiss-owns
+	@install -Dm755 contrib/kiss-size ${DESTDIR}${PREFIX}/bin/kiss-size
+	@install -Dm755 contrib/kiss-depends ${DESTDIR}${PREFIX}/bin/kiss-depends
```
Comment (dylanaraps) [inline Makefile]: This could be a `for` loop no?

File: Makefile [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+	@rm -f ${DESTDIR}${PREFIX}/bin/kiss-export
+	@rm -f ${DESTDIR}${PREFIX}/bin/kiss-maintainer
+	@rm -f ${DESTDIR}${PREFIX}/bin/kiss-manifest
+	@rm -f ${DESTDIR}${PREFIX}/bin/kiss-manifest-tree
+	@rm -f ${DESTDIR}${PREFIX}/bin/kiss-orphans
+	@rm -f ${DESTDIR}${PREFIX}/bin/kiss-owns
+	@rm -f ${DESTDIR}${PREFIX}/bin/kiss-repodepends
+	@rm -f ${DESTDIR}${PREFIX}/bin/kiss-reset
+	@rm -f ${DESTDIR}${PREFIX}/bin/kiss-revdepends
+	@rm -f ${DESTDIR}${PREFIX}/bin/kiss-size
```
Comment (dylanaraps) [inline Makefile]: This could also be a `for` loop. We could generate the list at the top using a `glob` and use it in both the install/uninstall spots.

**Comment by dylanaraps** [comment]
Basically:

```sh
target:
    for util in contrib/kiss-* ; do \
        install bla bla "$util"
    done
```

**Comment by cemkeylan** [comment]
I made it into a for loop, but it still requires a list of items to exist.

**Pull request closed without merging**